qutip/core/data/convert.pyx

```diff
@@ -258,6 +267,11 @@ cdef class _to:
                 self.weight[(to_t, from_t)] = weight
                 self._convert[(to_t, from_t)] =\
                     _converter(convert[::-1], to_t, from_t)
+        for dtype in self.dtypes:
+            self.weight[(dtype, Data)] = 1.
+            self.weight[(Data, dtype)] = self.anydataweight
```

Comment:
The constant here is rather awkward.

Could we add an small implementation of an epsilon value? Something like:
```
class _Epsilon:
    def __repr__(self):
        return "EPSILON"

    def __eq__(self, other):
        if isinstance(other, _Epsilon):
            return True
        return NotImplemented

    def __add__(self, other):
        if isinstance(other, _Epsilon):
            return self
        return other

EPSILON = _Epsilon()
```
Then we can check whether a weight is all epsilons by doing `if weight == EPSILON`.

Alternatively, could we just check whether the weight is `0.0`?

Maybe there are some practical issues here I am not seeing yet.

Reply:
There are issues with using `0.0`, but `_Epsilon` should work fine.

Reply:
Could you write a sentence about the issues with `0.0` so that I understand a bit the issues?

Reply:
When there are both a specific specialization and a general one:
(`kron_dense` vs `kron_data`)
We want `kron_dense` to have a lower weight so it does not decide to use `kron_data` instead.
`kron_dense`'s weight is zero, so I used an arbitrary small number.

Reply:
Ah, that makes sense.